src/libmongoc/tests/TestSuite.c

```diff
@@ -165,6 +166,16 @@ TestSuite_Init (TestSuite *suite, const char *name, int argc, char **argv)
       } else if ((0 == strcmp ("-s", argv[i])) ||
                  (0 == strcmp ("--silent", argv[i]))) {
          suite->silent = true;
+      } else if ((0 == strcmp ("--ctest-run", argv[i]))) {
+         if (suite->ctest_run) {
+            test_error ("'--ctest-run' can only be specified once");
+         }
+         if (argc - 1 == 0) {
+            test_error ("'--ctest-run' requires an argument");
+         }
+         suite->flags |= TEST_NOFORK;
```

Comment:
To check, ctest runs each test within a separate process to isolate, right? I assume this is necessary to avoid an extra process spawn, or for ctest to capture output.

Reply:
Yes. Similarly to how `test-libmongoc` (without `-f`) spawns a given test in its own process (via `fork ()`) to handle process termination and capture output without interrupting the runner, ctest does the same for each registered test case.

Reply:
This is correct.